plugins/database/mongodb/connection_producer.go

```diff
@@ -187,3 +156,112 @@ func (c *mongoDBConnectionProducer) secretValues() map[string]interface{} {
 		c.Password: "[password]",
 	}
 }
+
+func (c *mongoDBConnectionProducer) getConnectionURL() (connURL string) {
+	connURL = dbutil.QueryHelper(c.ConnectionURL, map[string]string{
+		"username": c.Username,
+		"password": c.Password,
+	})
+	return connURL
+}
+
+func (c *mongoDBConnectionProducer) getWriteConcern() (opts *options.ClientOptions, err error) {
+	if c.WriteConcern == "" {
+		return nil, nil
+	}
+
+	input := c.WriteConcern
+
+	// Try to base64 decode the input. If successful, consider the decoded
+	// value as input.
+	inputBytes, err := base64.StdEncoding.DecodeString(input)
+	if err == nil {
+		input = string(inputBytes)
+	}
+
+	concern := &writeConcern{}
+	err = json.Unmarshal([]byte(input), concern)
+	if err != nil {
+		return nil, errwrap.Wrapf("error unmarshalling write_concern: {{err}}", err)
```

Comment:
just fyi we can now do `fmt.Errorf("....%w", err)`

Reply:
I'm not a fan of the new error wrapping logic in `fmt.Errorf`. Do you have a particular reason to change this since this was only moved out of the `Init` function? I haven't been able to discern a standard error-making pattern since I've seen `fmt.Errorf`, `errwrap` and `github.com/pkg/errors` all in the repo.

Reply:
`errwrap` has historically been used over `pkg/errors`. Switching to the Go 1.13 error wrapping has been approved/preferred and can be done as convenient.  No need to update it here.